plugins/easyimage/plugin.js

```diff
@@ -504,10 +504,50 @@
 		return !CKEDITOR.env.ie || CKEDITOR.env.version >= 11;
 	}
 
+	function addUploadButtonToToolbar( editor ) {
+		var hiddenUploadElement;
+		if ( !editor._.easyImageHiddenUploadElement ) {
+			// Element is not attached to DOM, but still it might be `virtually` clicked.
+			hiddenUploadElement = CKEDITOR.dom.element.createFromHtml( '<input data-cke-easyimage-hidden-upload="1" style="display:none;" type="file" accept="image/*" tabindex="-1" multiple="true">' );
+
+			hiddenUploadElement.$.addEventListener( 'change', function( evt ) {
+				if ( evt.target.files.length ) {
+					// Simulate paste event, to support all nice stuff from imagebase (e.g. loaders) (#1730).
+					editor.fire( 'paste', {
+						method: 'paste',
+						dataValue: '',
+						dataTransfer: new CKEDITOR.plugins.clipboard.dataTransfer( { files: evt.target.files } )
+					} );
+				}
+			} );
+
+			editor.on( 'destroy', function() {
+				hiddenUploadElement.removeAllListeners();
```

Comment:
This function does nothing if listeners are bound via native DOM `addEventListener` as our editor does not hold any references to them.